src/fandom-wirewax.plugin.js

```diff
@@ -0,0 +1,257 @@
+  this.WirewaxOverlayHideHandler = this.WirewaxOverlayHideHandler.bind(this);
+  this.WirewaxOverlayEventHandler = this.WirewaxOverlayEventHandler.bind(this);
+  this.startTimeUpdate = this.startTimeUpdate.bind(this);
+  this.stopTimeUpdate = this.stopTimeUpdate.bind(this);
+  this.hotspotEvent = false;
+  this.debouncePlayPause = false;
+
+  this.player.on("playlistItem", function(){
+    if (this.embedder) {
+
+      this.stopTimeUpdate();
+
+      // Dispose pre video interaction
+      try {
+        this.embedder.dispose();
+      } catch (error) {
+        console.log(error);
+      }
+    }
+
+    // Search JW media id
+    var mediaId =
+      this.player.getConfig().playlistItem.mediaid ||
+      this.player.getConfig().playlistItem.videoId;
+
+    // validate interaction
+    if (!mediaId) {
+      throw new TypeError("No JW media id is specified.");
+    }
+
+    fetch(
+      'https://edge-player.wirewax.com/jwPlayerData/' + mediaId + '.txt'
+    )
+    .then(function(response){
+      if (response.status !== 200) {
+        throw new Error("No vidId is mapped with this mediaid");
+      }
+
+      return response.json();
+    })
+    .then(function(data){
+      this.vidId = data;
+
+      // Inject SDK
+      return injectEmbedderSDK();
+    }.bind(this))
+    .then(this.setupEmbedder)
+    .then(this.registerEvents)
+    .catch(function(error) {
+      console.warn(error);
+    });
+  }.bind(this));
+}
+
+FandomWirewaxPlugin.prototype.setupEmbedder = function () {
+  if (!this.embedder) {
+    // Create a container
+    this.container = document.createElement("div");
+    this.container.classList.add("vjs-wirewax-container");
+    this.container.setAttribute(
+      "style",
+      "position: absolute; height: 100%; width: 100%; top: 0; pointer-events: none"
+    );
+    this.player.getContainer().appendChild(this.container);
+
+    // Initialize embedder
+    this.embedder = window.createWirewaxEmbedder();
+  }
+
+  // Create video-less WIREWAX player
+  this.embedder.createEl(this.container, {
+    isPlugin: true,
+    videoId: this.vidId,
+    rootId: this.rootId,
+  });
+
+  return this.embedder.ready();
+}
+
+FandomWirewaxPlugin.prototype.registerEvents = function () {
+  // Custom time sync handler
+  var HTML5VideoEl = this.player.getConfig().mediaElement;
+  this.setWIREWAXCurrentTime = function () {
+    this.embedder.setCurrentTime(HTML5VideoEl.currentTime);
+    this.animationId = window.requestAnimationFrame(
+      this.setWIREWAXCurrentTime
+    );
+  }.bind(this);
+
+  // Handle the delay caused by injecting script
+  var isPlaying = this.player.getState() === "playing";
+
+  if (isPlaying || this.autoPlay) {
+    this.startTimeUpdate();
+    this.embedder.play();
+  }
+
+  if (this.isPlayerRegistered) return;
+
+  window.dataLayer = window.dataLayer || [];
+
+  // Bind WIREWAX to JW player events
+  this.player.on("play", this.JWPlayHandler);
```

Comment:
Whats the difference between JWPlayHandler and the WirewaxPlayHandler? I know one is the player but I dont understand why we need to use both